### ISSUE TITLE
Migrate from Travis CI to Github Actions to run our CI 

### ISSUE THREAD

**Issue by mattab** [piwik/piwik#20179]
as per title

work done in https://github.com/matomo-org/matomo/pull/18679

## Steps

- Update Global action to deal with Missing dependencies in some builds
- Update console command to generate the file. So that it's easy to update the Travis files. So you can create github action file by using a command. so it's easy to push the github actions to all available plugins without manually coping the github action files 
- check if Github builds at least as fast as travis? and report the runtime here to compare
- we make it work for Matomo core 
  - and also for the other repositories under the team
  - and we provide basic instructions to colleagues on how to convert a plugin repo to github
- Update the guides that mention Travis to replace with Github and fix appropriately -> Delete any outdated content and write a very basic guide on how Github works
  - https://developer.matomo.org/guides/tests-travis-extended
  - https://developer.matomo.org/guides/tests-travis
- Document how to handle dependencies in a plugin
  - eg. how would it work for LDAP plugin for example "[LoginLdap](https://github.com/matomo-org/plugin-LoginLdap), for example, tests itself against a live LDAP server and thus needs to install and setup OpenLDAP on travis. To accomplish this, [LoginLdap](https://github.com/matomo-org/plugin-LoginLdap) adds extra steps to its generated .travis.yml file." -> how would we do this on github action? Yes, with a local action it's possible.

Notes:

- we don't fix random failing tests for now
- we do fix any tests that  consistently fail

**Comment by sgiehl** [comment]
GitHub test action is now running for `4.x-dev` branch. Travis tests will still run in parallel (except UI tests).
Will create another PR to remove travis tests (including the sumbodule) for `5.x-dev` branch. So support for travis ci will be dropped with Matomo 5.
